### DISCUSSION TITLE
Square Payments Integration

### DISCUSSION THREAD

**Discussion by JanieHuang** [medusajs/medusa#3156] [Feature Requests]
I want to be able to integrate and use Square payments as a plugin.
https://developer.squareup.com/reference/square/payments-api

I have a POS with Square with my in-person payments. Ideally, I would like to stick to one platform and also use Square for online transactions as well so I can pull sales reports from one place (Square)

**Comment by srindom** [comment]
Without knowing Square's API my guess would be that this would be very fast to integrate. There is a guide on how to build payment providers here: https://docs.medusajs.com/advanced/backend/payment/how-to-create-payment-provider/

**Reply by sheyaln** [reply]
I'm very much a beginner, but I'd be happy to take a shot at it if you or one of the other maintainers would be willing to nit pick my spaghetti code, and maybe provide some guidance on how to create effective documentation for it?

**Comment by dylanswartz** [comment]
+1 would love this integration

**Comment by hdev505** [comment]
Hi @Milksheyke  @srindom  @dylanswartz  Sorry for the interruption. I am a one of medusa.js fan. Now i am facing square payment plugin development issue even if i didn't ever develop Square payment ... (I had a review on the document of Square api endpoints and some other details, but now i can't match the square apis and medusa.js Provider children funtions) ... Please provide or guide me how can i handle it. (it would be nice if you provide the code bc i think it will be same on every side) 

also faced the front-end side : can't find the best solution for the square sdk ... trying myself 

Thank you for your attention. Looking forward to talk with you soon
